packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,250 @@
+  entityID: UUID,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: Manage pagination
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    ConsistentRead: true,
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': partitionKeyForEvent(entity, entityID),
+    },
+  }
+
+  logger.debug('Searching events by entity and entity ID. Query params: ', params)
+  const result = await dynamoDB.query(params).promise()
+  return (result.Items as Array<EventEnvelope>) ?? []
+}
+
+interface EventStoreKeys {
+  [eventsStoreAttributes.partitionKey]: string
+  [eventsStoreAttributes.sortKey]: string
+}
+
+async function searchEventsByEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entity: string,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: manage pagination
+  // Fist query the index
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    IndexName: eventsStoreAttributes.indexByEntity.name(config),
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.indexByEntity.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': partitionKeyForIndexByEntity(entity, 'event'),
+    },
+  }
+
+  logger.debug('Searching events by entity. Index query params: ', params)
+  const partialResult = await dynamoDB.query(params).promise()
+  const indexRecords = (partialResult.Items as Array<EventStoreKeys>) ?? []
+  return findEventsDataWithKeys(dynamoDB, config, logger, indexRecords)
+}
+
+async function searchEventsByType(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  type: string,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: manage pagination
+  // Fist query the index
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    IndexName: eventsStoreAttributes.indexByType.name(config),
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.indexByType.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': type,
+    },
+  }
+
+  logger.debug('Searching events by type. Index query params: ', params)
+  const partialResult = await dynamoDB.query(params).promise()
+  const indexRecords = (partialResult.Items as Array<EventStoreKeys>) ?? []
+  return findEventsDataWithKeys(dynamoDB, config, logger, indexRecords)
```

Comment:
Once we have queried the index, we got the primary keys of the table. Now we need to query the main table with those keys